tests/serverpod_test_server/test_integration/auth_core/auth_header_test.dart

```diff
@@ -0,0 +1,77 @@
+import 'dart:async';
+
+import 'package:serverpod/serverpod.dart';
+import 'package:serverpod_test_client/serverpod_test_client.dart';
+import 'package:serverpod_test_server/test_util/test_key_manager.dart';
+import 'package:serverpod_test_server/test_util/test_serverpod.dart';
+import 'package:test/test.dart';
+
+void main() async {
+  TestAuthKeyManager authKeyManager = TestAuthKeyManager();
+  var client = Client(
+    'http://localhost:8080/',
+    authenticationKeyManager: authKeyManager,
+  );
+  late Serverpod server;
+
+  group('Passing auth key in HTTP header format', () {
+    late Completer<String> tokenInspectionCompleter;
+
+    setUp(() async {
+      tokenInspectionCompleter = Completer();
+
+      Future<AuthenticationInfo?> authenticationHandler(
+        Session session,
+        String token,
+      ) {
+        tokenInspectionCompleter.complete(token);
+        return Future.value(AuthenticationInfo(1, {}));
+      }
+
+      server = IntegrationTestServer.create(
+        authenticationHandler: authenticationHandler,
+      );
+      await server.start();
+    });
+
+    tearDown(() async {
+      await server.shutdown(exitProcess: false);
+    });
+
+    test(
+        'then endpoint method should receive plain auth key (i.e. in original format)',
+        () async {
+      var key = 'username-4711:password-4711';
+      authKeyManager.put(key);
+
+      var reflectedKey = await client.reflection.reflectAuthenticationKey();
+      expect(reflectedKey, key);
+
+      var receivedToken = await tokenInspectionCompleter.future;
+      expect(receivedToken, key);
+    });
```

Comment:
Suggestion: Break out to two separate tests.

Why always try to validate a single case in each test. For scenarios in e2e cases this is not always possible, but where it is possible we always aim to keep a test as narrow and focused as possible.

In this test, we both validate what the endpoint received (from the session) and what the authentication handler received. This is not correctly reflected in the description. Please break this out into two different tests.